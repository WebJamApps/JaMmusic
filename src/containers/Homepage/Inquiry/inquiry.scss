.tableSection {
    border: none; text-align: left; margin: 0; padding: 0; margin-bottom: 20px;
  }
