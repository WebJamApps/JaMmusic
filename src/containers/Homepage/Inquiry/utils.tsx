import superagent from 'superagent';
import type { IinquiryFormData } from '.';

function handleCountryChange(
  event: React.ChangeEvent<HTMLSelectElement>,
  formData:IinquiryFormData,
  setFormData:(arg0:IinquiryFormData)=>void,
): void {
  setFormData({ ...formData, country: event.target.value });
}

async function createEmailApi(
  emailForm: IinquiryFormData,
  setHasSubmitted:(arg0:boolean)=>void,
): Promise<void> {
  try {
    await superagent.post(`${process.env.BackendUrl}/inquiry`)
      .set('Content-Type', 'application/json')
      .send(emailForm);
    setHasSubmitted(true);
  } catch (e) { console.log((e as Error).message); }
}

async function handleSubmit(
  evt: React.MouseEvent<HTMLButtonElement, MouseEvent>,
  formData:IinquiryFormData,
  setHasSubmitted:(arg0:boolean)=>void,
): Promise<void> {
  const {
    firstname, lastname, emailaddress, uSAstate, country, phonenumber, zipcode, comments,
  } = formData;
  const emailForm = {
    firstname,
    lastname,
    emailaddress,
    uSAstate,
    country,
    phonenumber,
    zipcode,
    comments: (comments).trim(),
  };
  await createEmailApi(emailForm, setHasSubmitted);
}

function handleInputChange(
  evt: React.ChangeEvent<HTMLInputElement>,
  formData:IinquiryFormData,
  setFormData:(arg0:IinquiryFormData)=>void,
): string {
  const { id, value } = evt.target;
  setFormData({ ...formData, [id]: value });
  return value;
}

function handleUsStateChange(
  evt: React.ChangeEvent<HTMLSelectElement>,
  formData: IinquiryFormData,
  setFormData:(arg0:IinquiryFormData)=>void,
): void {
  setFormData({ ...formData, uSAstate: evt.target.value });
}

function continueValidating(formData:IinquiryFormData): boolean {
  const {
    country, uSAstate, firstname, lastname, zipcode, comments,
  } = formData;
  let validState = false, notEmpty = false;
  if (country === 'United States' && uSAstate !== '* Select Your State') validState = true;
  if (country !== 'United States') validState = true;
  if (firstname && lastname && zipcode && comments) notEmpty = true;
  if (notEmpty && country !== '* Select Your Country' && validState) {
    return false;
  }
  return true;
}

function validateForm(formData:IinquiryFormData): boolean {
  const {
    emailaddress,
  } = formData;
  // eslint-disable-next-line no-useless-escape, prefer-regex-literals
  const regEx = RegExp('^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$');
  if (regEx.test(emailaddress) && emailaddress.includes('.')) {
    return continueValidating(formData);
  } return true;
}

export default {
  handleCountryChange, handleSubmit, handleInputChange, handleUsStateChange, validateForm, createEmailApi, continueValidating,
};
