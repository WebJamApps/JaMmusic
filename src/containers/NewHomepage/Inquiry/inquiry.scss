.tableSection {
    border: none; text-align: left; padding: 0; margin-bottom: 20px; width: 90%; margin-top: 8px;
  }
  .contact-us{
    height: 500px !important;
    box-shadow: 0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.4);
  }
  .submit-inquiry{
    margin-right: 10px !important;
  }

