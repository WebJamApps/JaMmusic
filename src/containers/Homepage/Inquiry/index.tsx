// import { Component } from 'react';
// import superagent from 'superagent';
import {
  // TextareaAutosize,
  Button,
} from '@mui/material';
import { useState } from 'react';
// import { WjInput } from 'src/components/WjInput';
// import { WjDropdown } from 'src/components/WjDropdown';
// import stateData from './StateData.json';
// import countryData from './CountryData.json';
import utils from './utils';

export interface IinquiryFormData {
  firstname: string; lastname: string; emailaddress: string; uSAstate: string; country: string;
  phonenumber: string; zipcode: string; comments: string;
}

// export interface InquiryState {
//   country: string;
//   uSAstate: string;
//   firstname: string;
//   lastname: string;
//   zipcode: string;
//   comments: string;
//   emailaddress: string;
//   formError: string;
//   phonenumber: string;
//   submitted: boolean;
//   [x: number]: number;
// }

// export const isFormValid = (state: InquiryState) => {
//   const { formError } = state;
//   return formError !== '';
// };

// export function CommentsSection(
//   props: { currentState: InquiryState; comments: string; setState: (args0: InquiryState) => void; validateForm: () => void; },
// ) {
//   const {
//     comments, setState, validateForm, currentState,
//   } = props;
//   return (
//     <TextareaAutosize
//       style={{ marginTop: '20px', height: '80px', width: '100%' }}
//       placeholder="Comments"
//       className="comments"
//       value={comments}
//       onChange={(evt) => { setState({ ...currentState, comments: evt.target.value }); validateForm(); }}
//     />
//   );
// }

interface IformActionsProps {
  setHasSubmitted: (arg0: boolean) => void,
  formData:IinquiryFormData,
}
export function FormActions(props: IformActionsProps) {
  const { setHasSubmitted, formData } = props;
  // const { createEmail, currentState } = props;
  return (
    <div className="inquiryValidation input-field col" style={{ marginBottom: '12px' }}>
      <span className="inquiryValidation">* Required</span>
      <Button
        id="sendEmailButton"
        // disabled={isFormValid(currentState)}
        onClick={(evt) => {
          utils.handleSubmit(evt, formData, setHasSubmitted);
        }}
      >
        Submit
      </Button>
    </div>
  );
}

// onChange(evt: React.ChangeEvent<HTMLSelectElement>, isSelected?: boolean): void {
//   if (isSelected) {
//     this.setState({ uSAstate: evt.target.value });
//   } else { this.setFormField(evt.target.id, evt.target.value); }
//   this.validateForm();
// }

// onInputChange(evt: React.ChangeEvent<HTMLInputElement>): void {
//   this.setFormField(evt.target.id, evt.target.value);
//   this.validateForm();
// }

// setFormField(id: string, value: string): void { return this.setState((preS) => ({ ...preS, [id]: value.trim() })); }

// continueValidating(validEmail: boolean): boolean {
//   const {
//     country, uSAstate, firstname, lastname, zipcode, comments,
//   } = this.state;
//   let validState = false, notEmpty = false;
//   if (country === 'United States' && uSAstate !== '* Select Your State') validState = true;
//   if (country !== 'United States') validState = true;
//   if (firstname !== '' && lastname !== '' && zipcode !== '' && comments !== '') notEmpty = true;
//   if (notEmpty && validEmail && country !== '* Select Your Country' && validState) {
//     this.setState({ formError: '' });
//     return false;
//   }
//   this.setState({ formError: 'Complete all required fields' });
//   return true;
// }

// validateForm(): boolean {
//   const {
//     emailaddress,
//   } = this.state;
//   let validEmail = false;
//   // eslint-disable-next-line no-useless-escape, prefer-regex-literals
//   const regEx = RegExp('^[a-zA-Z0-9.!#$%&â€™*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$');
//   if (regEx.test(emailaddress) && emailaddress.includes('.')) {
//     validEmail = true;
//   } else {
//     if (emailaddress !== '') this.setState({ formError: 'Invalid email format' });
//     return true;
//   }
//   return this.continueValidating(validEmail);
// }

// interface ItableSectionProps {
//   firstname:string, lastname:string,
//   emailaddress:string,
//   phonenumber:string,
// }
// function TableSection(props:ItableSectionProps): JSX.Element {
//   const {
//     firstname, lastname,
//     emailaddress,
//     phonenumber,
//   } = props;
//   return (
//     <table style={{
//       border: 'none', textAlign: 'left', margin: 0, padding: 0, marginBottom: '20px',
//     }}
//     >
//       <tbody>
//         <tr className="white-background">
//           <td style={{ border: 'none', padding: 0 }}>
//             <WjInput label="First Name" isRequired type="text" onChange={this.onInputChange} value={firstname} />
//           </td>
//           <td style={{ border: 'none', padding: '8px' }}>{' '}</td>
//           <td style={{ border: 'none', padding: 0 }}>
//             <WjInput label="Last Name" isRequired type="text" onChange={this.onInputChange} value={lastname} />
//           </td>
//         </tr>
//         <tr className="white-background"><td style={{ border: 'none', padding: '8px' }}>{' '}</td></tr>
//         <tr className="white-background">
//           <td style={{ border: 'none', padding: 0 }}>
//             <WjInput label="Email Address" isRequired type="email" onChange={this.onInputChange} value={emailaddress} />
//           </td>
//           <td style={{ border: 'none', padding: '8px' }}>{' '}</td>
//           <td className="phone" style={{ border: 'none', padding: 0 }}>
//             {' '}
//             <WjInput label="Phone Number" isRequired type="tel" onChange={this.onInputChange} value={phonenumber} />
//           </td>
//         </tr>
//       </tbody>
//     </table>
//   );
// }
interface IinquiryFormProps {
  setHasSubmitted: (arg0: boolean) => void,
  setFormData: (arg0: IinquiryFormData) => void,
  formData: IinquiryFormData,
}
function InquiryForm(props: IinquiryFormProps): JSX.Element {
  const {
    formData,
    // setFormData,
    setHasSubmitted,
  } = props;
  // const {
  //   country,
  //   uSAstate,
  //   zipcode,
  //   comments, firstname, lastname, emailaddress, phonenumber,
  // } = formData;
  return (
    <form id="new-contact" className="col">
      {/* <TableSection
        firstname={firstname}
        lastname={lastname}
        emailaddress={emailaddress}
        phonenumber={phonenumber}
      /> */}
      {/* <WjDropdown
        style={{ width: '100%' }}
        htmlFor="country"
        onChange={(evt) => utils.handleCountryChange(evt, formData, setFormData)}
        value={country}
        options={this.countryValues}
      />
      {country === 'United States'
        ? (
          <WjDropdown
            htmlFor="state"
            onChange={this.onChange}
            value={uSAstate}
            options={this.stateValues}
          />
        )
        : <> </>}
      <p style={{ margin: 0 }}>&nbsp;</p>
      <WjInput
        label="Zipcode"
        isRequired
        type="text"
        onChange={this.onInputChange}
        value={zipcode}
        style={{ width: '100%' }}
      />
      <p style={{ margin: 0 }}>&nbsp;</p>
      <CommentsSection
        currentState={this.state}
        comments={comments}
        setState={this.setState}
        validateForm={this.validateForm}
      />
      <p className="form-errors" style={{ color: 'red', marginBottom: '-15px' }}>
        {formError}
      </p>
      */}
      <FormActions setHasSubmitted={setHasSubmitted} formData={formData} />
    </form>
  );
}

export function Inquiry(): JSX.Element {
  // const { submitted } = this.state;
  const [hasSubmitted, setHasSubmitted] = useState(false);
  const [formData, setFormData] = useState({} as IinquiryFormData);
  return (
    <div className="row form-row" style={{ margin: 'auto', border: '1px solid black' }}>
      {!hasSubmitted ? (
        <div className="contact-form">
          <h4 style={{
            textAlign: 'center', marginBottom: '0', marginTop: '10px', paddingTop: 0, fontWeight: 'bold',
          }}
          >
            Contact Us
          </h4>
          <InquiryForm setHasSubmitted={setHasSubmitted} formData={formData} setFormData={setFormData} />
        </div>
      ) : (
        <div className="page-content contacted">
          <p style={{ textAlign: 'center', margin: '14px', paddingBottom: '15px' }}>
            Thank you for contacting us.
            <br />
            We will respond to your request soon.
          </p>
        </div>
      )}
    </div>
  );
}
// }
