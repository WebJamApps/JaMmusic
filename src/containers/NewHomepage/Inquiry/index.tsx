import {
  TextareaAutosize,
  Button,
} from '@mui/material';
import { useState } from 'react';
import { WjInput } from 'src/components/WjInput';
import { WjDropdown } from 'src/components/WjDropdown';
import stateData from './StateData.json';
import countryData from './CountryData.json';
import utils from './utils';
import './inquiry.scss';

export interface IinquiryFormData {
  firstname: string; lastname: string; emailaddress: string; uSAstate: string; country: string;
  phonenumber: string; zipcode: string; comments: string;
}
interface IcommentsSectionProps {
  formData: IinquiryFormData, setFormData: (arg0: IinquiryFormData) => void
}
export function CommentsSection(props: IcommentsSectionProps) {
  const {
    formData, setFormData,
  } = props;
  const { comments } = formData;
  return (
    <TextareaAutosize
      style={{ marginTop: '20px', height: '80px', width: '100%' }}
      placeholder="* Comments"
      className="comments"
      value={comments}
      onChange={(evt) => {
        const { target: { value } } = evt;
        setFormData({ ...formData, comments: evt.target.value });
        return value;
      }}
    />
  );
}

interface IformActionsProps {
  setHasSubmitted: (arg0: boolean) => void,
  formData: IinquiryFormData,
}
export function FormActions(props: IformActionsProps) {
  const { setHasSubmitted, formData } = props;
  return (
    <div className="inquiryValidation input-field col" style={{ marginBottom: '12px' }}>
      <span className="inquiryValidation">* Required</span>
      <Button
        className="submit-inquiry"
        variant="contained"
        id="sendEmailButton"
        disabled={utils.validateForm(formData)}
        onClick={(evt) => {
          utils.handleSubmit(evt, formData, setHasSubmitted);
        }}
      >
        Submit
      </Button>
    </div>
  );
}
interface ItableSectionProps {
  formData: IinquiryFormData, setFormData: (arg0: IinquiryFormData) => void
}
export function EmailPhoneRow(props: ItableSectionProps) {
  const { formData, setFormData } = props;
  const { emailaddress, phonenumber } = formData;
  return (
    <tr className="white-background">
      <td style={{ border: 'none', padding: 0 }}>
        <WjInput
          label="Email Address"
          isRequired
          type="email"
          value={emailaddress}
          onChange={(evt) => utils.handleInputChange(evt, formData, setFormData)}
        />
      </td>
      <td style={{ border: 'none', padding: '8px' }}>{' '}</td>
      <td className="phone" style={{ border: 'none', padding: 0 }}>
        {' '}
        <WjInput
          label="Phone Number"
          isRequired
          type="tel"
          value={phonenumber}
          onChange={(evt) => {
            const { target: { value } } = evt;
            utils.handleInputChange(evt, formData, setFormData);
            return value;
          }}

        />
      </td>
    </tr>
  );
}

export function TableSection(props: ItableSectionProps): JSX.Element {
  const { formData, setFormData } = props;
  const { firstname, lastname } = formData;
  return (
    <table className="tableSection">
      <tbody>
        <tr className="white-background">
          <td style={{ border: 'none', padding: 0 }}>
            <WjInput
              label="First Name"
              isRequired
              type="text"
              value={firstname}
              onChange={(evt) => utils.handleInputChange(evt, formData, setFormData)}
            />
          </td>
          <td style={{ border: 'none', padding: '8px' }}>{' '}</td>
          <td style={{ border: 'none', padding: 0 }}>
            <WjInput
              label="Last Name"
              isRequired
              type="text"
              value={lastname}
              onChange={(evt) => {
                const { target: { value } } = evt;
                utils.handleInputChange(evt, formData, setFormData);
                return value;
              }}
            />
          </td>
        </tr>
        <tr className="white-background"><td style={{ border: 'none', padding: '8px' }}>{' '}</td></tr>
        <EmailPhoneRow formData={formData} setFormData={setFormData} />
      </tbody>
    </table>
  );
}

interface IinquiryFormProps {
  setHasSubmitted: (arg0: boolean) => void,
  setFormData: (arg0: IinquiryFormData) => void,
  formData: IinquiryFormData,
  staticCountry?:string
}
export function InquiryForm(props: IinquiryFormProps): JSX.Element {
  // eslint-disable-next-line object-curly-newline
  const { formData, setFormData, setHasSubmitted, staticCountry } = props;
  const { country, uSAstate, zipcode } = formData;
  return (
    <form id="new-contact" className="col">
      <TableSection formData={formData} setFormData={setFormData} />
      <WjDropdown
        disabled={!!staticCountry}
        style={{ width: '100%' }}
        htmlFor="country"
        value={staticCountry || country}
        onChange={(evt) => {
          const { target: { value } } = evt;
          utils.handleCountryChange(evt, formData, setFormData);
          return value;
        }}
        options={countryData.sort()}
      />
      {country === 'United States' || staticCountry === 'United States'
        ? (
          <WjDropdown
            htmlFor="state"
            value={uSAstate}
            onChange={(evt) => {
              const { target: { value } } = evt;
              utils.handleUsStateChange(evt, formData, setFormData);
              return value;
            }}
            options={stateData.sort()}
          />
        )
        : <> </>}
      <p style={{ margin: 0 }}>&nbsp;</p>
      <WjInput
        label="Zipcode"
        isRequired
        type="text"
        value={zipcode}
        onChange={(evt) => {
          const { target: { value } } = evt;
          utils.handleInputChange(evt, formData, setFormData);
          return value;
        }}
        style={{ width: '100%' }}
      />
      <p style={{ margin: 0 }}>&nbsp;</p>
      <CommentsSection formData={formData} setFormData={setFormData} />
      <FormActions setHasSubmitted={setHasSubmitted} formData={formData} />
    </form>
  );
}

export const ContactRequest = () => (
  <div className="page-content contacted">
    <p style={{ textAlign: 'center', margin: '14px', paddingBottom: '15px' }}>
      Thank you for contacting us.
      <br />
      We will respond to your request soon.
    </p>
  </div>
);
interface IcontactFormProps {
  hasSubmitted: boolean,
  hideTitle?: boolean,
  country?: string,
  setHasSubmitted: (arg0: boolean) => void
  formData: IinquiryFormData,
  setFormData: (arg0: IinquiryFormData) => void
}

export function ContactForm(props: IcontactFormProps) {
  const {
    hasSubmitted, hideTitle, country, setHasSubmitted, formData, setFormData,
  } = props;
  return (
    <div
      className="contact-us"
      style={{
        margin: 'auto',
      }}
    >
      {!hasSubmitted ? (
        <div className="contact-form">
          <h4 style={{
            textAlign: 'center', marginBottom: '0', marginTop: '1px', paddingTop: 0, fontWeight: 'bold',
          }}
          >
            {hideTitle ? '' : 'Contact Us'}
          </h4>
          <InquiryForm staticCountry={country} setHasSubmitted={setHasSubmitted} formData={formData} setFormData={setFormData} />
        </div>
      ) : (
        <ContactRequest />
      )}
    </div>
  );
}
export function Inquiry({ country, hideTitle }:{ country?:string, hideTitle?:boolean }): JSX.Element {
  const [hasSubmitted, setHasSubmitted] = useState(false);
  const [formData, setFormData] = useState({} as IinquiryFormData);
  const props = {
    hasSubmitted, hideTitle, country, setHasSubmitted, formData, setFormData,
  };
  return (
    <ContactForm {...props} />
  );
}

