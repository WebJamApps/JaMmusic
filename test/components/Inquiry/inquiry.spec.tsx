import React from 'react';
import { shallow } from 'enzyme';
import Inquiry from '../../../src/components/inquiry';

describe('Inquiry Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow<Inquiry>(<Inquiry />);
  });
  it('is defined', () => {
    expect(Inquiry).toBeDefined();
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('displays the usa state dropdown when country selected is United States', () => {
    wrapper.instance().setState({ country: 'United States' });
    const dropdown = wrapper.find('select#state').get(0);
    dropdown.props.onChange({ target: { value: 'Alaska' } });
    expect(wrapper.instance().state.uSAstate).toBe('Alaska');
  });
  it('calls on change for text input', () => {
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.hi).toBe('11'); });
    wrapper.instance().onChange({ target: { id: 'hi', value: '11 ' } });
  });
  it('calls on change for uSAstate dropdown', () => {
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.uSAstate).toBe('Alaska'); });
    wrapper.instance().onChange({ target: { value: 'Alaska' } }, true);
  });
  it('calls handleChange for country dropdown', () => {
    wrapper.instance().setState = jest.fn((obj) => { expect(obj.country).toBe('Spain'); });
    wrapper.instance().handleChange({ target: { value: 'Spain' } }, true);
  });
  it('checks for invalid phone numbers', () => {
    wrapper.setState({
      phonenumber: '540',
      validPhoneNumber: false,
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(true);
  });
  it('returns the validation', () => {
    wrapper.setState({
      firstname: 'Bob',
      lastname: 'McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      zipcode: '24179',
      phonenumber: '5405555555',
      validPhoneNumber: true,
      validEmail: true,
      formError: '',
      country: 'United States',
      uSAstate: 'Alaska',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(false);
  });
  it('returns the validation after fixing email format', () => {
    wrapper.setState({
      firstname: 'Bob',
      lastname: 'McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      zipcode: '24179',
      phonenumber: '5405555555',
      validPhoneNumber: true,
      validEmail: true,
      formError: 'Invalid email format',
      country: 'United States',
      uSAstate: 'Alaska',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(false);
  });
  it('returns the validation when bad email format', () => {
    wrapper.setState({
      firstname: 'Bob',
      lastname: 'McBobPerson',
      emailaddress: 'example-example.com',
      comments: 'A comment',
      zipcode: '24179',
      phonenumber: '5405555555',
      validPhoneNumber: true,
      validEmail: true,
      formError: '',
      country: 'United States',
      uSAstate: 'Alaska',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(true);
  });
  it('returns the validation when bad phone number', () => {
    wrapper.setState({
      firstname: 'Bob',
      lastname: 'McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      zipcode: '24179',
      phonenumber: '540',
      validPhoneNumber: true,
      validEmail: true,
      formError: '',
      country: 'United States',
      uSAstate: 'Alaska',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(true);
  });
  it('returns the validation when form not complete', () => {
    wrapper.setState({
      firstname: 'Bob',
      lastname: 'McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      zipcode: '24179',
      phonenumber: '5405555555',
      validPhoneNumber: true,
      validEmail: true,
      formError: 'Ten-digit phone number',
      country: 'United States',
      uSAstate: '--',
    });
    const result = wrapper.instance().validateForm();
    expect(result).toBe(true);
  });
  it('calls the email API', () => {
    wrapper.instance().setState = jest.fn((obJ) => { expect(obJ.submitted).toBe(true); });
    wrapper.instance().createEmailApi({ submitted: true });
  });
  it('creates an email', async () => {
    wrapper.instance().superagent.post = jest.fn(() => ({ set: () => ({ send: () => Promise.resolve({ status: 200 }) }) }));
    wrapper.update();
    wrapper.setState({
      fullname: 'Bob McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      uSAstate: 'Virginia',
    });
    const result = await wrapper.instance().createEmail();
    expect(result).toBe(200);
  });
  it('catches error when posting email to backend', async () => {
    wrapper.instance().superagent.post = jest.fn(() => ({ set: () => ({ send: () => Promise.reject(new Error('bad')) }) }));
    wrapper.update();
    wrapper.setState({
      fullname: 'Bob McBobPerson',
      emailaddress: 'example@example.com',
      comments: 'A comment',
      uSAstate: 'Virginia',
    });
    await expect(wrapper.instance().createEmail()).rejects.toThrow('bad');
  });
  it('calls the thank you statement', () => {
    wrapper.setState({
      submitted: true,
    });
    expect(wrapper.find('div.contacted').exists()).toBe(true);
  });
  it('calls handle change to set the comments', () => {
    wrapper.instance().handleChange({ target: { value: 'howdy' } });
    expect(wrapper.instance().state.comments).toBe('howdy');
  });
});
