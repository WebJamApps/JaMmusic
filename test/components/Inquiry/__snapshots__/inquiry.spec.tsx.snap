// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inquiry Form renders correctly 1`] = `ShallowWrapper {}`;
